taichi/common/serialization.h

```diff
@@ -636,9 +638,25 @@ class TextSerializer : public Serializer {
     this->process(key, t);
   }
 
+  // Entry to make an AOT json file
+  template <typename T>
+  void serialize_to_json(const char *key, const T &t) {
```

Comment:
Hmm, is this all we need for generating conforming JSON... ?

Reply:
Since our output structure is kinda stable so yea it's pretty simple to generate a conforming JSON. To simplify the problem, I didn't use `[]` at all, but keep using `{}` for now which is sufficient for simple cases. :P